libs/ui-components/src/components/common/OrganizationSelector.tsx

```diff
@@ -0,0 +1,205 @@
+import * as React from 'react';
+import {
+  ActionList,
+  ActionListGroup,
+  ActionListItem,
+  Bullseye,
+  Button,
+  Card,
+  CardBody,
+  CardTitle,
+  Flex,
+  FlexItem,
+  Menu,
+  MenuContent,
+  MenuItem,
+  MenuList,
+  PageSection,
+  Stack,
+  StackItem,
+  Text,
+  TextContent,
+  Title,
+} from '@patternfly/react-core';
+import { Modal, ModalBody, ModalHeader } from '@patternfly/react-core/next';
+
+import { Organization } from '@flightctl/types';
+import { useTranslation } from '../../hooks/useTranslation';
+import { useOrganizationGuardContext } from './OrganizationGuard';
+import { ORGANIZATION_STORAGE_KEY } from '../../utils/organizationStorage';
+
+interface OrganizationSelectorContentProps {
+  defaultOrganizationId?: string;
+  organizations: Organization[];
+  onSelect: (orgId: string) => void;
+  onCancel?: () => void;
+  allowCancel?: boolean;
+  isFirstLogin?: boolean;
+}
+
+const MAX_ORGANIZATIONS_FOR_SCROLL = 4;
+
+const OrganizationSelectorContent = ({
+  defaultOrganizationId,
+  organizations,
+  onSelect,
+  onCancel,
+  allowCancel = false,
+  isFirstLogin = false,
+}: OrganizationSelectorContentProps) => {
+  const { t } = useTranslation();
+  const [selectedOrg, setSelectedOrg] = React.useState<string | undefined>(defaultOrganizationId);
+  const hasUserMadeSelection = React.useRef(false);
+  const needsScroll = organizations.length > MAX_ORGANIZATIONS_FOR_SCROLL;
+
+  // Only set default organization if user hasn't made a manual selection yet
+  React.useEffect(() => {
+    if (defaultOrganizationId && !hasUserMadeSelection.current && !selectedOrg) {
+      setSelectedOrg(defaultOrganizationId);
+    }
+  }, [defaultOrganizationId, selectedOrg]);
+
+  const handleOrgSelection = React.useCallback((orgId: string) => {
+    hasUserMadeSelection.current = true;
+    setSelectedOrg(orgId);
+  }, []);
```

Comment:
Could this be simplified to the following..? 

```suggestion
  const [selectedOrg, setSelectedOrg] = React.useState<string>();
  const needsScroll = organizations.length > MAX_ORGANIZATIONS_FOR_SCROLL;

  // Only set default organization if user hasn't made a manual selection yet
  React.useEffect(() => {
    if (defaultOrganizationId && !selectedOrg) {
      setSelectedOrg(defaultOrganizationId);
    }
  }, [defaultOrganizationId, selectedOrg]);

  const handleOrgSelection = React.useCallback((orgId: string) => {
    setSelectedOrg(orgId);
  }, []);
```

Reply:
I actually removed the effect. I remember at some point there was a timing issue that defaultORg wasn't being received when the component mounted initially, but I couldn't see it happening anymore.